arduino-ide-extension/src/node/arduino-firmware-uploader-impl.ts

```diff
@@ -66,15 +70,27 @@ export class ArduinoFirmwareUploaderImpl implements ArduinoFirmwareUploader {
   }
 
   async flash(firmware: FirmwareInfo, port: string): Promise<string> {
-    return await this.runCommand([
-      'firmware',
-      'flash',
-      '--fqbn',
-      firmware.board_fqbn,
-      '--address',
-      port,
-      '--module',
-      `${firmware.module}@${firmware.firmware_version}`,
-    ]);
+    this.serialService.uploadInProgress = true;
+    await this.serialService.disconnect();
```

Comment:
Maybe, you can move this into the `try` block. If `disconnect` fails, the IDE won't stuck in the `uploadInProgress` state. What do you think? Just a proposal, though.

Reply:
Makes sense! Thanks, @kittaakos!